### PR TITLE
[DRAFT] Add emscripten_thread_pool_size to get thread pool size

### PR DESCRIPTION
**Pull request by axinging**
Currently no test case is added.

### CODE REVIEW

File: src/library_pthread.js [hunk 1]
```diff
@@ -660,6 +660,18 @@ var LibraryPThread = {
     return navigator['hardwareConcurrency'];
   },
 
+  emscripten_thread_pool_size: function() {
+#if PTHREAD_POOL_SIZE
+    const pthreadPoolSize = {{{ PTHREAD_POOL_SIZE }}};
+    if (pthreadPoolSize > _emscripten_num_logical_cores()) {
+      err('PTHREAD_POOL_SIZE is bigger than emscripten_num_logical_cores()!');
+    }
```
Comment (kripken) [inline src/library_pthread.js:668]: the warning seems too unrelated to me. that is, if this just gets the pool size, it shouldn't also validate it. if we think that warning is important, it should be present where the pool is created (and that could be in a separate PR).

File: src/library_pthread.js [hunk 2]
```diff
@@ -660,6 +660,18 @@ var LibraryPThread = {
     return navigator['hardwareConcurrency'];
   },
 
+  emscripten_thread_pool_size: function() {
+#if PTHREAD_POOL_SIZE
```
Comment (kripken) [inline src/library_pthread.js:664]: I think the entire body could be `return {{{ PTHREAD_POOL_SIZE || 0 }}}`.

(`{{{ }}}` evaluates at compile time, and the default value of that setting is the empty string)

**Review by kripken — COMMENTED** [review]
Seems like a reasonable addition. Thanks!

Please add a test (can grep to see where other similar functions are tested, like `emscripten_num_logical_cores`), and add yourself to AUTHORS.

**Review by sbc100 — COMMENTED** [review]
PTHREAD_POOL_SIZE is only the initial size of the pool.  Since the pool can grow over perhaps it would be more useful to return: `PThread.unusedWorkers.length()`?   That would give you the current number of available threads in the pool.  Or you could return `PThread.unusedWorkers.length() + PThread.runningWorkers.length()` which would five you the absolute size of the pool.
